### PR TITLE
Bump to contour 1.26 and use gateway-provisioner

### LINKED ISSUE #495: Do not use contour-operator
https://github.com/projectcontour/contour-operator is archived and 1.25+ is not available.
The script still uses operator to deploy Contour so we need to fix them.

### LINKED ISSUE #536: Bump Contour to v1.26
https://github.com/projectcontour/contour/releases/

### PR DESCRIPTION
**Pull request by ReToCode**
# Changes
- Bump to contour 1.26 
- Drop installation using contour-operator
- Use gateway-provisioner to deploy gateways
- Enable tests that now work: `websocket,websocket/split,host-rewrite`

/kind enhancement

Fixes #495
Fixes #536

**Release Note**

```release-note
net-gateway-api now tests with contour version 1.26 which supports websockets and host-rewrites
```

### CODE REVIEW

File: third_party/contour/gateway-internal.yaml [hunk 1]
```diff
@@ -15,35 +15,14 @@
...
-spec:
-  gatewayControllerName: projectcontour.io/contour-internal/contour
-  namespace:
-    name: contour-internal
-  ingressClassName: contour-internal
-  networkPublishing:
-    envoy:
-      containerPorts:
-      - name: http
-        portNumber: 8081
```
Comment (ReToCode) [inline third_party/contour/gateway-internal.yaml:36]: @dprotaso @nak3 how was this used? Seems like the gateway below just used port 80?

Not sure if we can define this using the gateway-provisioner. If this is needed, we probably need to use helm to have more configuration flags.

Comment (ReToCode) [reply third_party/contour/gateway-internal.yaml:36]: @KauzClay

Comment (nak3) [reply third_party/contour/gateway-internal.yaml:36]: IIRC, the `8081` port is used for prober. The current implementation lists endpoint IP which is `8081`, and probe against it.
If it works with private (cluster local) service works without `8081`, I think we can drop it, though.

**Comment by ReToCode** [comment]
>  message: 'service "ingress-conformance-host-rewrite-ragyzvdm" is invalid: serving-tests/ingress-conformance-host-rewrite-ragyzvdm
        is an ExternalName service, these are not currently enabled. See the config.enableExternalNameService
        config file setting, service "ingress-conformance-host-rewrite-ragyzvdm" is
        invalid: serving-tests/ingress-conformance-host-rewrite-ragyzvdm is an ExternalName
        service, these are not currently enabled. See the config.enableExternalNameService
        config file setting'

**Comment by nak3** [comment]
Deploying the changed manifest seems deploy `envoy-knative-local-gateway` svc with `LoadBalancer` not `ClusterIP`. Is that alright for internal envoy?

```
$ kubectl get svc -n contour-internal 
NAME                            TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
contour-knative-local-gateway   ClusterIP      10.96.50.182    <none>        8001/TCP       19m
envoy-knative-local-gateway     LoadBalancer   10.96.204.218   <pending>     80:30715/TCP   19m
```

**Comment by ReToCode** [comment]
Good point, seems to be the default. I added configuration for `ClusterIP`.

**Comment by nak3** [comment]
LGTM
@dprotaso @KauzClay will you or VMware team take a look?

**Review by dprotaso — APPROVED** [review]

**Pull request merged**